app/components/button_component.html.erb

```diff
@@ -1,13 +1,13 @@
 <%= container do %>
   <% if icon && (icon_position != :right) %>
-    <%= lucide_icon(icon, class: icon_classes) %>
+    <%= helpers.icon(icon, class: icon_classes) %>
```

Comment:
I think we may need to double-check both this `button_component` and `link_component` to make sure the classes we're passing through don't introduce any conflicts.  I think maybe the ideal solution is that our components pass through valid opts to the icon helper, such as:

```erb
<%= helpers.icon(icon, size: size, color: choose_based_on_variant_fn) %>
```

That would also allow us to _remove_ these icon dimensions from the `buttonish_component.rb` config and just use the sizes that the icon helper defines:

https://github.com/maybe-finance/maybe/blob/441f436187231ae4a0a696d353dcb38bbbf1eed0/app/components/buttonish_component.rb#L43

Reply:
Good feedback, thanks! I think I've implemented what you were looking for now. In theory we could have some kind of separate `icon_size` on the buttonish components, but I kind of like that icon size being tied to the size of the button itself, which should help ensure nice dimensions too. Let me know if my update wasn't quite what you were looking for.